compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -646,4 +647,21 @@ object TreeChecker {
       tp
     }
   }.apply(tp0)
+
+  /** Check that the tree only contains legal children trees */
```

Comment:
This documentation is a bit too general. I would also state the exact rule that is checked.
```
/** Check that the tree only contains legal children trees 
 *  
 *    - TypeTree can only appear in TypeApply args, New, Typed tpt, ...
 */
```